crates/cli/src/subcommands/generate/rust.rs

```diff
@@ -851,42 +837,45 @@ impl __sdk::spacetime_module::InModule for Reducer {{
                 },
                 "}\n",
             );
-        },
-        "}\n",
-    );
-
-    out.delimited_block(
-        "impl TryFrom<__ws::ReducerCallInfo<__ws::BsatnFormat>> for Reducer {",
-        |out| {
-            writeln!(out, "type Error = __anyhow::Error;");
             out.delimited_block(
-                "fn try_from(value: __ws::ReducerCallInfo<__ws::BsatnFormat>) -> __anyhow::Result<Self> {",
-                    |out| {
-                        out.delimited_block(
-                            "match &value.reducer_name[..] {",
-                            |out| {
-                                for reducer in iter_reducers(module) {
-                                    writeln!(
-                                        out,
-                                        "{:?} => Ok(Reducer::{}(__sdk::spacetime_module::parse_reducer_args({:?}, &value.args)?)),",
-                                        reducer.name.deref(),
-                                        reducer_variant_name(&reducer.name),
-                                        reducer.name.deref(),
-                                    );
-                                }
+                "fn parse_call_info(
+                    reducer_id_to_name: &impl Fn(__ws::ReducerId) -> anyhow::Result<&'static str>,
+                    raw: __ws::ReducerCallInfo<__ws::BsatnFormat>,
+                ) -> anyhow::Result<Self> {",
+                |out| {
+                    out.delimited_block(
+                        "use __sdk::spacetime_module::parse_reducer_args;
```

Comment:
I generally prefer to avoid `use`-importing definitions within generated code, as I'm wary of name conflicts. In this specific case I don't see any possibility of a conflict, but just as a rule I prefer not thinking about whether it's even possible.

Don't change this, I'm just musing. In the future I may do some revisions to this file and add a file-level doc comment with a style guide.